tests/pipeline/test_quarters_estimates.py

```diff
@@ -0,0 +1,955 @@
+                              q1_knowledge,
+                              q2_knowledge,
+                              comparable_date):
+        return pd.DataFrame()
+
+    def test_estimates(self):
+        dataset = QuartersEstimates(1)
+        engine = SimplePipelineEngine(
+            lambda x: self.loader,
+            self.trading_days,
+            self.asset_finder,
+        )
+        results = engine.run_pipeline(
+            Pipeline({c.name: c.latest for c in dataset.columns}),
+            start_date=self.trading_days[1],
+            end_date=self.trading_days[-2],
+        )
+        for sid in self.ASSET_FINDER_EQUITY_SIDS:
+            sid_estimates = results.xs(sid, level=1)
+            ts_sorted_estimates = self.events[
+                self.events[SID_FIELD_NAME] == sid
+            ].sort(TS_FIELD_NAME)
+            for i, date in enumerate(sid_estimates.index):
+                comparable_date = date.tz_localize(None)
+                # Filter out estimates we don't know about yet.
+                ts_eligible_estimates = ts_sorted_estimates[
+                    ts_sorted_estimates[TS_FIELD_NAME] <= comparable_date
+                ]
+                # If there are estimates we know about:
+                if not ts_eligible_estimates.empty:
+                    # Determine the last piece of information we know about
+                    # for q1 and q2. This takes advantage of the fact that we
+                    # only have 2 quarters in the test data.
+                    q1_knowledge = ts_eligible_estimates[
+                        ts_eligible_estimates[FISCAL_QUARTER_FIELD_NAME] == 1
+                    ]
+                    q2_knowledge = ts_eligible_estimates[
+                        ts_eligible_estimates[FISCAL_QUARTER_FIELD_NAME] == 2
+                    ]
+                    expected_estimate = self.get_expected_estimate(
+                        q1_knowledge,
+                        q2_knowledge,
+                        comparable_date,
+                    )
+                    # Have to explicitly check for None because
+                    # `expected_estimate` might be a DataFrame.
+                    if expected_estimate is not None:
+                        assert_series_equal(
+                            sid_estimates.iloc[i],
+                            expected_estimate[sid_estimates.columns],
+                            check_names=False
+                        )
+                    else:
+                        # There are no eligible 'next'/'previous' estimates on
+                        # this day; everything should be null.
+                        assert_true(sid_estimates.iloc[i].isnull().all())
+                else:
+                    # We don't know about any estimates on this day;
+                    # everything should be null.
+                    assert_true(sid_estimates.iloc[i].isnull().all())
+
+
+class NextEstimate(WithEstimatesTimeZero, ZiplineTestCase):
```

Comment:
Hmm, not seeing that one my machine. Thought it could be because I've changed that test around.